### ISSUE TITLE
False-positive for `legacy_objc_type` rule

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#3555]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The rule seems to match anything that **starts** with the [hard-coded names](https://github.com/realm/SwiftLint/commit/eca9ea4bb8bfeb4db640af409bf31101184c5c24#diff-4a54730b34dd348d489825a0ee3fa743d7c4d38271aed57a85c3ae27f79c2acbR5-R29), causing false-positives. It should only match on the full identifier name.

For example, this causes a warning because `NSData` is in the list:

```swift
try? NSDataDetector(types: 0)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.43.0

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Installer

* Paste your configuration file:

```yml
only_rules:
  - legacy_objc_type
```

**Comment by PH9** [comment]
I have a problem with 2 of `NS` classes
- `NSNumber` that needed to be used with `NSFormatter`.
- `NSDictionary` needed for `XCTAssertEqual` in unit testing.

Hope it can exclude or custom. Should I open another issue or leave it here?

**Comment by sindresorhus** [comment]
> Should I open another issue or leave it here?

It should be another issue. It's unrelated to this issue.
